az-cvm-vtpm/src/vtpm/mod.rs

```diff
@@ -111,12 +112,17 @@ pub enum QuoteError {
     NotAQuote,
     #[error("Wrong signature, that should not occur")]
     WrongSignature,
+    #[error("PCR bank not found")]
+    PcrBankNotFound,
+    #[error("PCR reading error")]
+    PcrRead,
 }
 
 #[derive(Serialize, Deserialize, Debug, Clone)]
 pub struct Quote {
     pub signature: Vec<u8>,
     pub message: Vec<u8>,
+    pub pcrs: Vec<Vec<u8>>,
```

Comment:
Sadly those changes will break semver, so we need to bump the crate to v0.5. we should also add a `#[non_exhaustive]` flag to both error enums (QuoteError and VerifyError) so we can extend them without breaking semver later. We can also remove the `pub` from the fields of the Quote struct, for similar reasons. If we refactor the internal data representation we don't necessarily break the API

Reply:
BTW why do we have two versions of `VerifyError`? One in `mod.rs` https://github.com/kinvolk/azure-cvm-tooling/blob/b353d341bbcfc032ff4d5d20f694c6cee290a7f0/az-cvm-vtpm/src/vtpm/mod.rs#L171-L182 and one in `verify.rs` https://github.com/kinvolk/azure-cvm-tooling/blob/b353d341bbcfc032ff4d5d20f694c6cee290a7f0/az-cvm-vtpm/src/vtpm/verify.rs#L10-L22

Reply:
good point that's a bug, since we have to bump to version 0.5, we can remove the one on vtpm/mod.rs (see #42)

Reply:
With this change we cannot access `message` any more: https://github.com/kinvolk/azure-cvm-tooling/blob/b353d341bbcfc032ff4d5d20f694c6cee290a7f0/az-cvm-vtpm/az-snp-vtpm/src/main.rs#L77

Should I add getter methods?

Reply:
yes, since we have quote.nonce() already, we can also add quote.message()

Reply:
done.